### PR TITLE
Introduce limits on the number of events allowed per command context

### LINKED ISSUE #39: CosmosDb TransactionalBatch limitations
Persisting a large batch of events in a single command throws an exception that doesn't clearly explain why the operation failed

**CosmosDb Transactional batch operation limitations**
Currently, there are two known limits:
- The Azure Cosmos DB request size limit constrains the size of the TransactionalBatch payload to not exceed 2 MB, and the maximum execution time is 5 seconds.
- There's a current limit of 100 operations per TransactionalBatch to ensure the performance is as expected and within SLAs.

**Expected behavior**
Throw a more concrete error message in a specialized exception type would be the preferred way to describe why the operation failed

### PR DESCRIPTION
**Pull request by christianhelle**
The changes here introduce hard limits to the number of events the system can accept per operation.

Each command operation will consist of at least 2 operations in a transactional batch, one for upserting the stream metadata and the other for adding events.

The following limitations are introduced:
- A maximum of 10 events per command context (CQRS)
- A maximum of 50 events per stream batch (Event Store)

**CosmosDb Transactional batch operation limitations**
Currently, there are two known limits:
- The Azure Cosmos DB request size limit constrains the size of the TransactionalBatch payload to not exceed 2 MB, and the maximum execution time is 5 seconds.
- There's a current limit of 100 operations per TransactionalBatch to ensure the performance is as expected and within SLAs.

This resolves #39

### CODE REVIEW

**Comment by christianhelle** [comment]
@LarsSkovslund I wrote this last night and thought that we can use this as a starting point for implementation detail discussions

**Review by LarsSkovslund — CHANGES_REQUESTED** [review]
Please remember to also update the change log.

File: src/Atc.Cosmos.EventStore/Events/EventLimitExceededException.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System.Diagnostics.CodeAnalysis;
+
+namespace Atc.Cosmos.EventStore.Events;
+
+[SuppressMessage("Design", "CA1032:Implement standard exception constructors", Justification = "By Design")]
+[SuppressMessage("Major Code Smell", "S3925:\"ISerializable\" should be implemented correctly", Justification = "By Design")]
+public class EventLimitExceededException : EventStoreException
```
[the remark this answers is not in the document]
Comment (christianhelle) [reply src/Atc.Cosmos.EventStore/Events/EventLimitExceededException.cs:7]: Yeah, it should be enough to use [`ArgumentOutOfRangeException`](https://learn.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception) so I'll do that. 

I originally thought that [`ArgumentOutOfRangeException`](https://learn.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception) was specific for collection indexers so I just checked the docs and I can verify that it doesn't seem to be specific to indexers

[the remark this answers is not in the document]
Comment (christianhelle) [reply src/Atc.Cosmos.EventStore/Events/EventLimitExceededException.cs:7]: After diving into the code and looking at the docs for [choosing the right standard exception](https://learn.microsoft.com/en-us/dotnet/api/system.exception?view=net-7.0#Standard) I think it best to use [`InvalidOperationException`](https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception) instead. 

`ArgumentOutOfRangeException` is designed to be used for the **argument** passed in, asserting the value of the argument itself

[the remark this answers is not in the document]
Comment (LarsSkovslund) [reply src/Atc.Cosmos.EventStore/Events/EventLimitExceededException.cs]: agreed, please remember to update the `CHANGELOG.md`

[the remark this answers is not in the document]
Comment (christianhelle) [reply src/Atc.Cosmos.EventStore/Events/EventLimitExceededException.cs]: I updated the change log

**Review by LarsSkovslund — APPROVED** [review]

**Review by LarsSkovslund — APPROVED** [review]

**Pull request merged**